processing/src/main/java/io/druid/query/groupby/epinephelinae/RowBasedGrouperHelper.java

```diff
@@ -420,30 +409,54 @@ public RowBasedKey fromByteBuffer(ByteBuffer buffer, int position)
       }
 
       if (fudgeTimestamp == null) {
-        return new Grouper.KeyComparator()
-        {
-          @Override
-          public int compare(ByteBuffer lhsBuffer, ByteBuffer rhsBuffer, int lhsPosition, int rhsPosition)
+        if (sortByDimsFirst) {
+          return new Grouper.KeyComparator()
           {
-            final int timeCompare = Longs.compare(lhsBuffer.getLong(lhsPosition), rhsBuffer.getLong(rhsPosition));
-            if (timeCompare != 0) {
-              return timeCompare;
+            @Override
+            public int compare(ByteBuffer lhsBuffer, ByteBuffer rhsBuffer, int lhsPosition, int rhsPosition)
+            {
+              for (int i = 0; i < dimCount; i++) {
```

Comment:
Maybe extract this + the next 8 lines into a method, which could be used in both comparators, for DRY and self-documenting?